libtcmu.c

```diff
@@ -325,6 +335,54 @@ static void close_devices(struct tcmulib_context *ctx)
 	}
 }
 
+static int reconfig_device(struct tcmulib_context *ctx, char *dev_name, char *cfgstring)
+{
+	struct tcmu_device **dev_ptr;
+	struct tcmu_device *dev;
+	int i = 0;
+	bool found = false;
+	char *reason = NULL;
+
+	darray_foreach(dev_ptr, ctx->devices) {
+		dev = *dev_ptr;
+		size_t len = strnlen(dev->dev_name, sizeof(dev->dev_name));
+		if (strncmp(dev->dev_name, dev_name, len)) {
+			i++;
+		} else {
+			found = true;
+			break;
+		}
+	}
+
+	if (!found) {
+		tcmu_err("could not reconfigure device %s: not found\n",
+			 dev_name);
+		goto err_free;
+	}
+
+	dev->handler = find_handler(ctx, dev->cfgstring);
+	if (!dev->handler) {
+		tcmu_err("could not find handler for %s\n", dev->dev_name);
+		goto err_free;
+	}
+
+	/* Handle all configuration changes in check_config */
+	if (dev->handler->check_config &&
```

Comment:
I think we can put all the new code in check_config? 

So if new configuration is found it would update based upon each individual handler, then flush if it needs to / reopen?

Reopen should probably be a new callout?